src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -356,11 +368,18 @@ public override void OnStart()
 				var dialog = Dialog;
 
 				if (dialog is null || dialog.Window is null || View is null)
+				{
+					// SAFETY: Fire event even on early return to prevent deadlock
+					FirePresentationCompleted();
 					return;
+				}
 
 				int width = ViewGroup.LayoutParams.MatchParent;
 				int height = ViewGroup.LayoutParams.MatchParent;
 				dialog.Window.SetLayout(width, height);
+
+				// Signal that the modal is fully presented and ready
+				FirePresentationCompleted();
```

Comment:
From what I investigated, the best place to add Fire it is on `OnResume`. `OnStart` will complete before the page is presented on the screen.